File: src/test/java/no/novari/flyt/mapping/service/ValueConvertingReferenceServiceTest.java
package no.novari.flyt.mapping.service;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class ValueConvertingReferenceServiceTest {

    private final ValueConvertingReferenceService service = new ValueConvertingReferenceService();

    @Test
    void testGetFirstValueConverterId_ValidInput() {
        String mappingString = "something $vc{123} something";

        Long result = service.getFirstValueConverterId(mappingString);

        assertEquals(123L, result);
    }

    @Test
    void testGetFirstValueConverterId_NoValueConvertingReference() {
        String mappingString = "something {123} something";

        Exception exception = assertThrows(IllegalArgumentException.class, () ->
                service.getFirstValueConverterId(mappingString)
        );

        String expectedMessage = "Mapping string contains no valid value converting reference";
        String actualMessage = exception.getMessage();

        assertEquals(expectedMessage, actualMessage);
    }

    @Test
    void testGetFirstValueConverterId_InvalidNumber() {
        String mappingString = "something $vc{abc} something";

        Exception exception = assertThrows(NumberFormatException.class, () ->
                service.getFirstValueConverterId(mappingString)
        );

        String expectedMessage = "For input string: \"abc\"";
        String actualMessage = exception.getMessage();

        assertEquals(expectedMessage, actualMessage);
    }
}

File: src/test/kotlin/no/novari/flyt/mapping/service/ValueConvertingReferenceServiceTest.kt
package no.novari.flyt.mapping.service

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test

class ValueConvertingReferenceServiceTest {
    private val service = ValueConvertingReferenceService()

    @Test
    fun testGetFirstValueConverterId_ValidInput() {
        val mappingString = "something \$vc{123} something"

        val result = service.getFirstValueConverterId(mappingString)

        assertEquals(123L, result)
    }

    @Test
    fun testGetFirstValueConverterId_NoValueConvertingReference() {
        val mappingString = "something {123} something"

        val exception =
            assertThrows(IllegalArgumentException::class.java) {
                service.getFirstValueConverterId(mappingString)
            }

        val expectedMessage = "Mapping string contains no valid value converting reference"
        val actualMessage = exception.message

        assertEquals(expectedMessage, actualMessage)
    }

    @Test
    fun testGetFirstValueConverterId_InvalidNumber() {
        val mappingString = "something \$vc{abc} something"

        val exception =
            assertThrows(NumberFormatException::class.java) {
                service.getFirstValueConverterId(mappingString)
            }

        val expectedMessage = "For input string: \"abc\""
        val actualMessage = exception.message

        assertEquals(expectedMessage, actualMessage)
    }
}
